plugins/kotlin/src/main/java/org/quiltmc/quiltflower/kotlin/KotlinWriter.java

```diff
@@ -125,6 +130,20 @@ private static boolean invokeProcessors(TextBuffer buffer, ClassNode node) {
       if (DecompilerContext.getOption(IFernflowerPreferences.DECOMPILE_ASSERTIONS)) {
         AssertProcessor.buildAssertions(node);
       }
+
+      List<KProperty> properties = new ArrayList<>();
+      Set<String> propertyFields = new HashSet<>();
+      Set<String> propertyAccessors = new HashSet<>();
+      boolean success = KProperty.parse(node, properties, propertyFields, propertyAccessors);
+      if (success) {
+        this.properties = properties;
+        this.propertyFields = propertyFields;
+        this.propertyAccessors = propertyAccessors;
+      } else {
```

Comment:
I think this whole part with the lists and sets need to be refactored. Why not assign them all to final arraylists/hashsets, use the fields in the parse call, and if !success then clear the lists/sets? You would probably have to clear them beforehand too. Or, does this need shared state in the first place? Can this be refactored to use a param and a class?